api/src/test/java/org/openmrs/module/ModuleFileParserUnitTest.java

```diff
@@ -166,31 +160,28 @@ public void parse_shouldFailIfModuleFileIsNull() {
 		String messageKey = "Module.error.fileCannotBeNull";
 		whenGettingMessageFromMessageSourceServiceWithKeyReturnSameKey(messageKey);
 
-		expectModuleExceptionWithMessage(messageKey);
-
-		parser.parse((File) null);
+		ModuleException exception =assertThrows(ModuleException.class, () -> parser.parse((File) null));
+		assertThat(exception.getMessage(), is(messageKey));
 	}
 
 	@Test
 	public void parse_shouldFailIfModuleFileHasInvalidExtension() {
 
-		String messageKey = "Module.error.invalidFileExtension";
+		String messageKey = "Module.error.invalidFileExtension Module: unknownmodule.jar";
```

Comment:
this is the problem :)

`whenGettingMessageFromMessageSourceServiceWithKeyReturnSameKey` sets up the mock. if you look at what it does its basically

```java
	private void whenGettingMessageFromMessageSourceServiceWithKeyReturnSameKey(String messageKey) {
		when(messageSourceService.getMessage(messageKey)).thenReturn(messageKey);
	}
```

The `messageKey` like `"Module.error.invalidFileExtension` will actually be passed to the `getMessage` in the implementation, which is why we are using it inside the Mockito `when` and just to make it simple in our assertion we also use it in setting up the return value of the mock `thenReturn`.

Since you changed the `messageKey` the mock is never called with `"Module.error.invalidFileExtension Module: unknownmodule.jar"` in the implementation and so it returns null (which is the default) :)

So what you need to do is leave the `messageKey` values as they were before your changes. Then in the assertion use `assertThat(exception.getMessage(), startsWith(messageKey));` (why see my comment before) and the tests will pass.

Reply:
thank you @teleivo

Reply:
there is  no Executable in junit 4.for now i have comment that method . and  i use

                ModuleException thrown = assertThrows(ModuleException.class, executable);
		assertThat(thrown.getMessage(), startsWith(expectedMessage));
directly in the test

Reply:
in Junit4 the parameter should be a ThrowingRunnable

so it becomes

```java
	private void expectModuleExceptionWithMessage(ThrowingRunnable executable, String expectedMessage) {
		ModuleException thrown = assertThrows(ModuleException.class, executable);
		assertThat(thrown.getMessage(), startsWith(expectedMessage));
	}
```

and tests that did use `expectModuleExceptionWithMessage` before use it like

```java
expectModuleExceptionWithMessage(() -> parser.parse(writeConfigXmlToFile(config)), messageKey);
```

Reply:
and please be aware of the whitespaces, I see

`ModuleException exception =assertThrows`

which should be `ModuleException exception = assertThrows`